.pre-commit-config.yaml

```diff
@@ -0,0 +1,53 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-json
+    -   id: check-added-large-files
+-   repo: https://github.com/python-jsonschema/check-jsonschema
+    rev: 0.13.0
+    hooks:
+    # Following only exists on non-EOL Python versions
+    #-   id: check-metaschema
+    #    name: "Check MSR Definition Format"
+    #    files: json_schemas/[^/]+\.schema\.json$
+    -   id: check-jsonschema
+        name: "Check MSR Definition Format"
+        files: IGNOREFORNOW/msr_[^/]+\.json$
+        args: ["--schemafile", "service/json_schemas/msrs.schema.json"]
+-   repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+    -   id: rst-backticks
+    -   id: rst-directive-colons
+    -   id: rst-inline-touching-normal
+-   repo: https://github.com/pycqa/flake8
+    rev: '5.0.4'
+    hooks:
+    -   id: flake8
+        args: ["--max-line-length=120", "--extend-ignore=E261"]
+-   repo: https://github.com/pycqa/bandit
+    rev: '1.7.0'
+    hooks:
+    -   id: bandit
+        # Only gate high-severity issues
+        args: ["-lll"]
```

Comment:
This checks for common security issues in python code.

It can be pretty nit-picky by default. Should we just gate on high-severity issues?

Also note that the pre-commit framework is *opt-in*, so this config file doesn't actually prevent these issues from getting committed. It's a convenience for committers to see issues when they introduce them instead of later during review or during our pre-release checklist.